pkg/collector/parser/receiver.go

```diff
@@ -155,9 +156,23 @@ func portName(receiverName string, port int32) string {
 }
 
 func portFromEndpoint(endpoint string) (int32, error) {
-	i := strings.LastIndex(endpoint, ":") + 1
-	part := endpoint[i:]
-	port, err := strconv.ParseInt(part, 10, 32)
+	var err error
+	var port int64
+
+	var r = regexp.MustCompile(":[0-9]+")
```

Comment:
nit: syntax `r := ....` is more used if the variable is initialized at creation

Reply:
Just fixed.